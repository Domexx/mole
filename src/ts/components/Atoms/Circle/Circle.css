/* === === === === Circle === === === === */

.Circle-enabled {
  cursor: pointer;
}

.Circle-disabled {
  cursor: not-allowed;
}

.Circle--green {
  background-color: rgb(76, 154, 56);
}

.Circle--red {
  background-color: rgb(179, 36, 36);
}

.Circle--yellow {
  background-color: rgb(217, 204, 18);
}

.Circle--gray {
  background-color: rgba(255, 255, 255, 0.337);
}
