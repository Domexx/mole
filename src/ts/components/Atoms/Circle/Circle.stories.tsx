import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import CircleComponent from './Circle';

const Component = {
  title: 'Components/Atoms/Circle',
  component: CircleComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof CircleComponent>> = (args) => {
  return <CircleComponent {...args} />;
};

export const Circle = Template.bind({});
export default Component;
