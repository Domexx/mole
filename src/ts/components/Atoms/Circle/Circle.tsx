import React, { useContext } from 'react';
import { GameContextData, GameContext } from '../../../states/GameStateProvider';

export type Props = {
  index: number;
  onPicked: (index: number) => void;
};

const Circle: React.FC<Props> = ({ index, onPicked }) => {
  const gameData: GameContextData = useContext(GameContext);

  const active = gameData.state.ActiveCircleIndex === index;
  const picked = gameData.state.PickedCircleIndex === index;
  const otherPicked = gameData.state.PickedCircleIndex && gameData.state.PickedCircleIndex !== index;
  const disabled = gameData.state.PickedCircleIndex;

  const getCircleStateClasses = () => {
    if (active) {
      if (picked) {
        return 'bg-green-300 border-green-600';
      }
      if (otherPicked) {
        return 'bg-red-200 border-red-500';
      }
      return 'bg-yellow-200 border-yellow-500';
    }
    if (picked) {
      return 'bg-gray-400 border-gray-700';
    }
    return 'bg-gray-300 border-gray-300';
  };

  const circleDisabledClasses = disabled ? 'cursor-not-allowed' : 'cursor-pointer';

  const renderContent = () => {
    if (active) {
      if (picked) {
        return <>ğŸ˜€</>;
      }
      if (otherPicked) {
        return <>ğŸ¤£</>;
      }
      return <>ğŸ™‚</>;
    }
    if (picked) {
      return <>ğŸ˜œ</>;
    }

    // eslint-disable-next-line react/jsx-no-useless-fragment
    return <></>;
  };

  return (
    <div
      onClick={() => {
        if (!disabled) {
          onPicked(index);
        }
      }}
      role="button"
      tabIndex={0}
      className={`w-12 h-12 flex items-center justify-center border-4 text-xl sm:w-20 sm:h-20 rounded-full ${getCircleStateClasses()} ${circleDisabledClasses}`}
      aria-hidden="true"
      data-testid="Circle"
    >
      {renderContent()}
    </div>
  );
};

export default Circle;
