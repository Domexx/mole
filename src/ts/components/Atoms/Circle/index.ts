import loadable from '@loadable/component';

import './Circle.css';

export default loadable(() => {
  return import('./Circle');
});
