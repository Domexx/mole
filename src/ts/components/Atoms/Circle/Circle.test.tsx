import React from 'react';
import { toJson } from '@test';
import renderer from 'react-test-renderer';
import Circle from './Circle';

test('Matches Snapshot', () => {
  const component = renderer.create(<Circle />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
