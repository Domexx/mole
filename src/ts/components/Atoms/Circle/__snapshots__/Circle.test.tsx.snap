// Vitest Snapshot v1

exports[`Matches Snapshot 1`] = `
<div
  aria-hidden="true"
  className="w-12 h-12 flex items-center justify-center border-4 text-xl sm:w-20 sm:h-20 rounded-full bg-green-300 border-green-600 cursor-pointer"
  data-testid="Circle"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  ðŸ˜€
</div>
`;
